codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -333,16 +358,21 @@ let createBatchInternal = (
         ~fetchStatesMap,
         ~arbitraryEventQueue,
         ~isUnorderedMultichainMode,
+        ~onlyBelowReorgThreshold,
       )
 
       isInReorgThresholdRef := isInReorgThresholdRef.contents || isInReorgThreshold
 
       switch val {
       | None => ()
       | Some({item, popItemOffQueue}) =>
-        popItemOffQueue()
-        batch->Js.Array2.push(item)->ignore
-        loop()
+        //For dynamic contract pre registration, allow creating a batch up to the reorg threshold
+        let shouldNotAddItem = isInReorgThreshold && onlyBelowReorgThreshold
+        if !shouldNotAddItem {
+          popItemOffQueue()
+          batch->Js.Array2.push(item)->ignore
+          loop()
+        }
```

Comment:
So we can create a full batch right up until the reorg threshold without throwing away any values.